exporter/snowflakeexporter/internal/utility/utility.go

```diff
@@ -0,0 +1,87 @@
+// Copyright observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package utility provides utility functions for the snowflakeexporter package to consolidate code
+package utility // "github.com/observiq/bindplane-agent/exporter/snowflakeexporter/internal/utility"
+
+import (
+	"encoding/hex"
+	"encoding/json"
+
+	"github.com/lib/pq"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.uber.org/zap"
+)
+
+// ConvertAttributesToString converts the pcommon.Map into a JSON string representation
+// this is due to a bug/lacking feature with the snowflake driver that prevents maps from being inserted into VARIANT & OBJECT columns
+// github issue: https://github.com/snowflakedb/gosnowflake/issues/217
+func ConvertAttributesToString(attributes pcommon.Map, logger *zap.Logger) string {
+	m := make(map[string]string, attributes.Len())
+	attributes.Range(func(k string, v pcommon.Value) bool {
+		m[k] = v.AsString()
+		return true
+	})
+
+	j, err := json.Marshal(m)
+	if err != nil {
+		logger.Warn("failed to marshal attribute map", zap.Error(err), zap.Any("map", m))
+		return ""
+	}
+	return string(j)
+}
+
+// TraceIDToHexOrEmptyString returns a string representation of the TraceID in hex
+func TraceIDToHexOrEmptyString(id pcommon.TraceID) string {
+	if id.IsEmpty() {
+		return ""
+	}
+	return hex.EncodeToString(id[:])
+}
+
+// SpanIDToHexOrEmptyString returns a string representation of the SpanID in hex
+func SpanIDToHexOrEmptyString(id pcommon.SpanID) string {
+	if id.IsEmpty() {
+		return ""
+	}
+	return hex.EncodeToString(id[:])
+}
+
+// FlattenExemplars will flatten the given exemplars into slices of the individual fields
+func FlattenExemplars(exemplars pmetric.ExemplarSlice, l *zap.Logger) (pq.StringArray, pq.StringArray, pq.StringArray, pq.StringArray, pq.Float64Array) {
```

Comment:
These functions don't have tests are we able to test them?

Reply:
Yea I definitely can. I didn't originally because they're from an [internal package](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/internal/coreinternal/traceutil/traceutil.go) in collector-contrib. Because they are in an internal pkg I couldn't import it, so I just copied them here since they're pretty simple

Reply:
Gotcha, lets copy the tests then and also just put a comment on here pointing out where you got them from.

Reply:
Gotcha, it doesn't look like they test them so I'll come up with a couple quick ones and add the comments